finality-aleph/src/lib.rs

```diff
@@ -157,8 +161,8 @@ pub fn session_id_from_block_num<B: Block>(num: NumberFor<B>, period: SessionPer
     SessionId(num.saturated_into::<u32>() / period.0)
 }
 
-pub struct AlephConfig<B: Block, N, C, SC> {
-    pub network: N,
+pub struct AlephConfig<B: Block, H: ExHashT, C, SC> {
+    pub network: Arc<NetworkService<B, H>>,
```

Comment:
Maybe rather than the specific `NetworkService` you could require an `N: network::Network + network::RequestBlocks<H>`? I believe this is what we actually want, right?

Reply:
It is not that simple :crying_cat_face: